### PR TITLE
[CALCITE-3601] Update elasticsearch tests upgrade from junit4 to junit5

### PR DESCRIPTION
**Pull request by XuQianJin-Stars**
As illustrated in CALCITE-3601
Update `elasticsearch` tests upgrade from junit4 to junit5.

### CODE REVIEW

File: elasticsearch/src/test/java/org/apache/calcite/adapter/elasticsearch/EmbeddedElasticsearchPolicy.java [hunk 1]
```diff
@@ -104,7 +98,7 @@ public static EmbeddedElasticsearchPolicy create() {
    *  }
    * </pre>
    *
-   * @param index index of the index
+   * @param index   index of the index
```
Comment (vlsi) [inline elasticsearch/src/test/java/org/apache/calcite/adapter/elasticsearch/EmbeddedElasticsearchPolicy.java]: Please refrain from aligning parameter descriptions.
It creates lots of unrelated changes, and it is prone to merge conflicts in the future.

See sample reasoning (~ `it creates problems for future maintenance` ): https://google.github.io/styleguide/javaguide.html#s4.6.3-horizontal-alignment

Comment (XuQianJin-Stars) [reply elasticsearch/src/test/java/org/apache/calcite/adapter/elasticsearch/EmbeddedElasticsearchPolicy.java]: > Please refrain from aligning parameter descriptions.
> It creates lots of unrelated changes, and it is prone to merge conflicts in the future.
> 
> See sample reasoning (~ `it creates problems for future maintenance` ): https://google.github.io/styleguide/javaguide.html#s4.6.3-horizontal-alignment

well, Let me change it.

**Comment by vlsi** [comment]
It turned out to be simpler than I initially thought.

I don't quite like that the PR includes javadoc re-formats for `EmbeddedElasticsearchPolicy` class, however, I'm inclined that it does not harm that much for `EmbeddedElasticsearchPolicy` because it is not that likely different PRs would edit the class.

File: elasticsearch/src/test/java/org/apache/calcite/adapter/elasticsearch/EmbeddedElasticsearchPolicy.java [hunk 2]
```diff
@@ -216,4 +209,11 @@ private TransportAddress httpAddress() {
     return node.httpAddress();
   }
 
+  @Override public void afterAll(ExtensionContext context) throws Exception {
```
Comment (vlsi) [inline elasticsearch/src/test/java/org/apache/calcite/adapter/elasticsearch/EmbeddedElasticsearchPolicy.java]: Just a small note: could you please move the methods to the location of previous `before` and `after`?

Comment (XuQianJin-Stars) [reply elasticsearch/src/test/java/org/apache/calcite/adapter/elasticsearch/EmbeddedElasticsearchPolicy.java]: Well , Ok Let me change it.  In addition, can we add style to the calite project to define the code format in calite?

Comment (vlsi) [reply elasticsearch/src/test/java/org/apache/calcite/adapter/elasticsearch/EmbeddedElasticsearchPolicy.java]: What do you mean by "code format"?

I'm inclined that we should prefer https://github.com/apache/calcite/blob/master/.editorconfig when possible.

Comment (vlsi) [reply elasticsearch/src/test/java/org/apache/calcite/adapter/elasticsearch/EmbeddedElasticsearchPolicy.java]: PS. please move `afterAll` as well

Comment (XuQianJin-Stars) [reply elasticsearch/src/test/java/org/apache/calcite/adapter/elasticsearch/EmbeddedElasticsearchPolicy.java]: hi @vlsi  Is your original idea to implement declarative registration extensions and kotlin?

Comment (vlsi) [reply elasticsearch/src/test/java/org/apache/calcite/adapter/elasticsearch/EmbeddedElasticsearchPolicy.java]: No, I just had no idea how to approach the test.

The approach you use seems to be nice. I have not checked TestContainer best practices yet. They have a case for "keeping test container up for multiple test classes", so there might be something to learn from.

Comment (XuQianJin-Stars) [reply elasticsearch/src/test/java/org/apache/calcite/adapter/elasticsearch/EmbeddedElasticsearchPolicy.java]: > No, I just had no idea how to approach the test.
> 
> The approach you use seems to be nice. I have not checked TestContainer best practices yet. They have a case for "keeping test container up for multiple test classes", so there might be something to learn from.

I am also interested in learning about "keeping test container up for multiple test classes".

Comment (vlsi) [reply elasticsearch/src/test/java/org/apache/calcite/adapter/elasticsearch/EmbeddedElasticsearchPolicy.java]: It looks like JUnit provides no solution yet, so TestContainers use their own: https://github.com/testcontainers/testcontainers-java/pull/887/files#diff-0dcf4d2cadeb3e526020477bc12e1420R73

Comment (XuQianJin-Stars) [reply elasticsearch/src/test/java/org/apache/calcite/adapter/elasticsearch/EmbeddedElasticsearchPolicy.java]: > It looks like JUnit provides no solution yet, so TestContainers use their own: https://github.com/testcontainers/testcontainers-java/pull/887/files#diff-0dcf4d2cadeb3e526020477bc12e1420R73

Well, indeed, TestContainers also looks good. Thank you very much for letting me learn something new.

**Pull request closed without merging**